### PR TITLE
Fix build warning

### PR DESCRIPTION
**Pull request by patrickkuo**
Closing https://github.com/MystenLabs/sui/issues/5663

### CODE REVIEW

File: crates/sui-core/src/execution_engine.rs [hunk 1]
```diff
@@ -459,7 +459,7 @@ fn pay<S>(
 ) -> Result<(), ExecutionError> {
     check_recipients(&recipients, &amounts)?;
     let (mut coins, coin_type) = check_coins(&coin_objects, None)?;
-    let (total_coins, total_amount) = check_total_coins(&coins, &amounts)?;
+    let (_total_coins, _total_amount) = check_total_coins(&coins, &amounts)?;
```
Comment (gegaowp) [inline crates/sui-core/src/execution_engine.rs:462]: I remember the warnings were gone already with `cargo build -r`, is it warning from some other cmds?

Comment (gegaowp) [reply crates/sui-core/src/execution_engine.rs:462]: The related PR is https://github.com/MystenLabs/sui/pull/5555

Comment (patrickkuo) [reply crates/sui-core/src/execution_engine.rs:462]: debug_assert_eq causes the warning, if the build is not in test/debug mode the variable used by `debug_assert_eq` will appear to be a unused variable

Comment (patrickkuo) [reply crates/sui-core/src/execution_engine.rs:462]: if the warning is gone already we can close this PR, I will double check later today

Comment (gegaowp) [reply crates/sui-core/src/execution_engine.rs:462]: yeah, i remember the unused warn was there when we had a debug flag but the debug_assert_eq macro seems working fine.

**Review by huitseeker — COMMENTED** [review]
You could instead not rename and do `#[cfg(debug_assertions)]` on top of your variable definitions which are only used in `debug_assert!` checks.

**Comment by patrickkuo** [comment]
I don't see the warnings anymore, I think this is fixed by https://github.com/MystenLabs/sui/pull/5555, closing this PR

**Pull request closed without merging**